modules/ding_campaign_plus/ding_campaign_plus.module

```diff
@@ -0,0 +1,395 @@
+      break;
+    }
+  }
+
+  // If any of the modules reports "false" a no-hit-trigger is found.
+  if ($triggers === FALSE) {
+    // Send notification to site administrator.
+    $campaign = node_load($campaign_nid);
+
+    $address = variable_get('site_mail', '');
+    if (!empty($address)) {
+      $to = $address;
+      $from = $address;
+      $body = array();
+      $body[] = t('Hi');
+      $body[] = '';
+      $body[] = t('The campaign "!title" will not be shown as it do not generate any hits in the data-well.', array('!title' => $campaign->title));
+      $body[] = '';
+      $body[] = t('The campaign can be found here !url', array('!url' => url('node/' . $campaign_nid, array('absolute' => TRUE))));
+
+      $params = array('body' => implode("\n", $body));
+      drupal_mail('ding_campaign_plus', 'notification', $to, language_default(), $params, $from);
+    }
+    else {
+      watchdog('ding_campaign_plus', 'Notification mail not set as site mail is empty', array(), WATCHDOG_NOTICE);
+    }
+  }
+}
+
+/**
+ * Implements hook_mail().
+ */
+function ding_campaign_plus_mail($key, &$message, $params) {
+  switch ($key) {
+    case 'notification':
+      $message['subject'] = t('Inactive campaign detected');
+      $message['body'][] = $params['body'];
+      break;
+  }
+}
+
+/**
+ * Implements hook_cron().
+ */
+function ding_campaign_plus_cron() {
+  $last_run = variable_get('ding_campaign_plus_cron_last_run', 0);
+
+  // Only add campaigns to validation queue on different intervals.
+  if (REQUEST_TIME - $last_run >= variable_get('ding_campaign_plus_notification', 604800) || $last_run == 0) {
+    $result = db_select('node', 'n')
+      ->fields('n', array('nid'))
+      ->condition('type', 'ding_campaign_plus')
+      ->condition('status', 1)
+      ->execute();
+
+    $queue = DrupalQueue::get('ding_campaign_plus_notification');
+    foreach ($result as $info) {
+      $queue->createItem($info->nid);
+    }
+
+    variable_set('ding_campaign_plus_cron_last_run', REQUEST_TIME);
+  }
+}
+
+/**
+ * Get information about the different triggers/rules available for campaigns.
+ *
+ * @see hook_ding_campaign_plus_info()
+ *
+ * @return array
+ *   Basic information about the triggers.
+ */
+function ding_campaign_plus_rules_info() {
+  $result = &drupal_static(__FUNCTION__);
+  if (!isset($result)) {
+    $result = [];
+    foreach (module_implements('ding_campaign_plus_info') as $module) {
+      $result[$module] = module_invoke($module, 'ding_campaign_plus_info');
+    }
+  }
+
+  return $result;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function ding_campaign_plus_form_ding_campaign_plus_node_form_alter(&$form, &$form_state) {
+  // Add states to the node form based on campaign type selection.
+  $form['field_ding_campaign_plus_text']['#states'] = array(
+    'invisible' => array(
+      ':input[name="field_ding_campaign_plus_type[und]"]' => array('value' => 'campaign_image'),
```

Comment:
wouldn't it be better to use a :input[data-state="foo"] selector?

Reply:
I don't know data-state? Can't find anything about it in FAPI or jquery docs?

Reply:
it's just an example data attribute, see: 
https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes

Reply:
yes, okay.... 

But it's an alter to the node edit form. So that would require that I also alter the default fields created by drupal core!